design-documents/service-isolation/caching-layer.md

```diff
@@ -0,0 +1,113 @@
+# Caching layer for Magento services
+
+## Overview
+
+Cache is one of the most important parts in a modern web application architecture.
+Magento caching layer should effectively improve:
+ * Performance
+ * Availability
+ * Scalability
+
+Caching should be implemented on several levels:
+ * Caching of http-responses returned by [BFF](https://github.com/magento/architecture/blob/master/design-documents/service-isolation.md#backends-for-frontends) (exposed web-server) 
+ * Application data caching (results of DB queries, merged configurations etc.)
+
+### Caching results of GET-requests to BFF (exposed web-server)
+
+[BFF](https://github.com/magento/architecture/blob/master/design-documents/service-isolation.md#backends-for-frontends) http-responses contains static assets, **public** and **private** dynamic content.
+Static assets should be cached using CDN. Reverse proxy should cache public content.
+It could be Varnish (acts as reverse proxy) or Fastly (combines reverse proxy and CDN functions).
+Varnish and Fastly are already integrated with Magento. Reverse proxy uses lazy caching approach.
+Beside performance this type of cache has next benefits:
+ * Protection against outages - can optionally serve stale content when there is a problem with origin server
+ * Scalability - number of caching nodes can be increased
+ * Flexibility – Varnish Configuration Language (VCL) builds customized solutions, rules and modules
+
+ ![Public content caching](img/public-cache.jpg)
```

Comment:
Note that these diagrams show GraphQL sitting behind a cache/proxy. This is fine, but note that an HTTP cache will _not_ actually handle caching of GraphQL requests/responses, as they're all done as a `POST`

Reply:
Thanks, added a note about it